server/handlers/json-manifest-handlers.js

```diff
@@ -0,0 +1,41 @@
+const get = require("lodash/get");
+const Promise = require("bluebird");
+
+exports.handleManifest = function handleManifest(req, res, next, {config, logError, manifestFn}) {
+  return new Promise(resolve => resolve(manifestFn(config)))
+    .then(result => {
+      res.setHeader("Cache-Control", "public,max-age=900");
+      res.setHeader("Vary", "Accept-Encoding");
+      res.json(Object.assign({
+        name: get(config, ["publisher-settings", "title"]),
+        short_name: get(config, ["publisher-settings", "title"]),
+        start_url: "/",
+        display: "standalone",
+        background_color: "#eee",
+      }, result));
+    }).catch(e => {
+      logError(e);
+      res.status(500);
+      res.send(e.message);
+    }).finally(() => res.end());
+};
+
+exports.handleAssetLink = function handleAssetLink(req, res, next, {config, logError, assetLinkFn}) {
+  return new Promise(resolve => resolve(assetLinkFn(config)))
```

Comment:
If you want to always resolve a Promise, you can instead use `Promise.resolve` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve